sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java

```diff
@@ -141,16 +139,17 @@ static MetricExporter configureOtlpMetrics(
 
   private static void configureIntervalMetricReader(
       ConfigProperties config, SdkMeterProvider meterProvider, MetricExporter exporter) {
-    IntervalMetricReaderBuilder readerBuilder =
-        IntervalMetricReader.builder()
-            .setMetricProducers(Collections.singleton(meterProvider))
-            .setMetricExporter(exporter);
+
     Duration exportInterval = config.getDuration("otel.imr.export.interval");
-    if (exportInterval != null) {
-      readerBuilder.setExportIntervalMillis(exportInterval.toMillis());
+    if (exportInterval == null) {
+      // TODO: What default for reading?
```

Comment:
In the current spec PR - "The default value is 5000 (milliseconds)."

Reply:
Wow, that's quite.... fast.   The default in OpenCensus was 1 minute.   Should we slow it down a bit?

Reply:
I don't have strong opinion, I got the value (5000ms) from the [tracing SDK spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#batching-processor) for consistency (and I think consistency here is not something super important).

Note the difference - for span processor, the default 5000ms is the **delay between two consecutive exports**, so the actual interval is something like 5000ms + _exporting time_, in metrics we are taking different approach by using a fixed schedule (so the delay is 5000ms - _exporting time_).

Reply:
I think there's also a huge difference in batch size + aggregation between metrics + traces.

Traces are known to be high-frequency + highly sampled.   Metrics we're leveraging aggregation to limit  bandwidth, so the export interval *should* be different.    Given what I know of our customers + vendor requests around metrics, I think a default of 1 minute should accommodate most users (and be too fast for some).

Reply:
I know that for Microsoft most of the systems use either 20 seconds or 1 minute so I'm totally fine with 1 min.

Reply:
I've updated the PR to change it to 60 seconds https://github.com/open-telemetry/opentelemetry-specification/pull/1888/commits/68f4fdc1156ae3a8eed3332006040b8648db32e1.

Reply:
Hm. Having to wait a full minute, on top of ingest time for the backend is a very long time to wait for metrics to appear on new deployments. What's the harm in having it be faster?

Reply:
In OpenCensus there was the agility to have a different delay between when the first metrics are exported and the average time between each export.

My thinking on this is as follows:

- For short-lived processes, the shutdown hook + export is crucial, and one of the benefits of push-metrics.
- For most other metrics, one of the reasons to use metrics OVER logs or traces is because you can greatly shrink the sheer size of the data.   If we report too frequently we put more pressure on backends to collapse the data or risk being very expensive for users.
- For scenarios where we're serving HTTP traffic, if we send too frequently, we're unlikely to have explored/opened all metric-streams within the reporting interval.  This actually has memory/gc implications for the implementation.
- For initial-startup "is it working" scenarios, it probably makes sense to default a quick fire of metrics

TL;DR:

- Our default should balance "Most metric streams have a point recorded during this time" and "Users don't run a huge bill on storage", "Users can see changes live" and "Nice for debugging"
- I think the "is it working on startup" use case is enough to shorten time-to-initial report, but not the on-going reporting interval.  There's some other nice benefits we may be able to get here like restart-loop detection.  However I think the cost of reporting too frequently is a real thing.  Users should opt-in if they want higher resolution.

Reply:
It would be really cool to have a dynamic system that would send metrics more frequently during unusual circumstances (like startup), but I have no idea how one might do that. :)